konig-data-app-common/src/main/java/io/konig/data/app/common/DataAppServlet.java

```diff
@@ -55,14 +69,48 @@ public void init() throws ServletException {
 			) {
 				yaml.addDeserializer(URI.class, new UriDeserializer());
 				dataApp = yaml.readObject(DataApp.class);
-				configure();
+				configure();
+				configAuthenticationDetails(null);
 			}
 		} catch (Throwable e) {
 			throw new ServletException(e);
 		}
 		
 	}
-
+
+	private static String convertToSHA1String(String value) throws ServletException {
+		try {			
+			MessageDigest  md = MessageDigest.getInstance("SHA-1");
+			byte[] bytes = md.digest(value.getBytes());
+			StringBuffer stringBuffer = new StringBuffer();
+	        for (int i = 0; i < bytes.length; i++) {
+	            stringBuffer.append(Integer.toString((bytes[i] & 0xff) + 0x100, 16)
+	                    .substring(1));
+	        }
+	        return stringBuffer.toString();
+		}catch(Throwable e) {
+			throw new ServletException(e);
+		}
+	}
+	
+	private void configAuthenticationDetails(String username) throws ServletException {
+		MemcacheService cache = MemcacheServiceFactory.getMemcacheService();
+		DatastoreOptions options = DatastoreOptions.newBuilder().build();
+		Query<?> query = Query.newGqlQueryBuilder("Select * from UserAuthentication").build();
+		Datastore datastore = options.getService();
+		QueryResults<?> results = datastore.run(query);
+		while (results.hasNext()) {
+			Entity currentEntity = (Entity) results.next();
+			cache.put(currentEntity.getString("username"), currentEntity.getString("password"));
```

Comment:
Consider renaming the "password" column to "passwordSha1" as a reminder that the value is a hash (not the raw value).